hooks/post_plan.py

```diff
@@ -127,8 +129,21 @@ def _validate_subnets(self, cache_subnet_group_name: str) -> str | None:
                 )
                 continue
             vpc_ids.add(subnet["VpcId"])
+
         if len(vpc_ids) > 1:
             self.errors.append("All subnets must belong to the same VPC")
+
+        # Check that all requested availability zones are covered by the subnet group
+        cache_group_subnet_availability_zones = {
+            s.get("SubnetAvailabilityZone", {}).get("Name") for s in cache_group_subnets
+        }
```

Comment:
`cache_group_subnet_availability_zones` can contain `None` because of conditional dict attr access, error message will show None in list too. Maybe real response always return non none values, to git rid of none values:

```python
        cache_group_subnet_availability_zones = {
            name
            for s in cache_group_subnets
            if (name := s.get("SubnetAvailabilityZone", {}).get("Name"))
        }
```

Reply:
I don't get it! How can `cache_group_subnet_availability_zones´ contain any `None`?

Reply:
the type of
```python
        cache_group_subnet_availability_zones = {
            s.get("SubnetAvailabilityZone", {}).get("Name") for s in cache_group_subnets
        }
```

will be `set[str | None]` because of those optional fields, for example, given this cache group subnet (if returned from AWS),

```python
{
  "SubnetAvailabilityZone": {},
}
```

Reply:
Got it. Changed in [24f93a9](https://github.com/app-sre/er-aws-elasticache/pull/373/commits/24f93a9d0d12114d29c77c1b487372618d5264e8)